### ISSUE TITLE
Select: value is always of type string

### ISSUE THREAD

**Issue by TuxujPes** [ant-design/ant-design#11155]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.6.5

### Environment
all

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/r4orqp3m0q)

### Steps to reproduce
- pass array of [DataSourceItemObject](https://github.com/ant-design/ant-design/blob/1bf0bab2a7bc0a774119f501806e3e0e3a6ba283/components/auto-complete/index.tsx#L11) to select
- click into input
- select any option

### What is expected?
`value` is of the same type as passed into Autocomplete

### What is actually happening?
`value` is of type `String` which may distort inputted data

**Comment by Gong-Elory** [comment]
read api docs  [onSelect](https://ant.design/components/auto-complete-cn/#API)  carefully!!!
the first param is what you need
![image](https://user-images.githubusercontent.com/22674497/42374722-a2056136-814b-11e8-95dc-d6b53be10a00.png)

**Comment by TuxujPes** [comment]
@Gong-Elory yes, you are correct. Sorry, seems like I've started explaining a problem from a very wrong angle.

So the *expected* `value` is of same type as it was passed

Seems like I went looking into "option" with anticipation to find some  input source data.

Select also accepts only string as `value` and `defaultValue`.
Probably this is conceptual limitation.

_Note: I've updated title and desc_
